src/Sentry/SessionUpdate.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Text.Json;
+using Sentry.Internal.Extensions;
+
+namespace Sentry
+{
+    /// <summary>
+    /// Snapshot of a session.
+    /// </summary>
+    public class SessionUpdate : IJsonSerializable
+    {
+        /// <summary>
+        /// Session.
+        /// </summary>
+        public Session Session { get; }
+
+        /// <summary>
+        /// Whether this is the initial snapshot.
+        /// </summary>
+        public bool IsInitial { get; }
+
+        /// <summary>
+        /// Timestamp.
+        /// </summary>
+        public DateTimeOffset Timestamp { get; }
+
+        /// <summary>
+        /// Duration of time since start of the session.
+        /// </summary>
+        public TimeSpan Duration => Timestamp - Session.Timestamp;
+
+        internal SessionUpdate(Session session, bool isInitial, DateTimeOffset timestamp)
+        {
+            Session = session;
```

Comment:
Isn't this risky? Now we're holding a reference to a mutable object.
Aren't we better off just copying all fields here instead?

Reply:
Yeah, good point